### PR TITLE
chore: bump arrow from 40.0.0 to 43.0.0.

### PR DESCRIPTION
**Pull request by youngsofun**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

Summary about this PR

- Closes #issue

### CODE REVIEW

**Comment by Xuanwo** [comment]
arrow2 seems depend on arrow 40?

**Review by sundy-li — APPROVED** [review]

**Comment by Xuanwo** [comment]
icelake also limit the version to `< 43`:

https://github.com/icelake-io/icelake/blob/00fffcacb928d64716343af83f684838a72ca62d/Cargo.toml#L19-L20

Would you like to bump the version range to `>=40, <44`?

**Comment by Xuanwo** [comment]
Thanks @youngsofun, https://github.com/icelake-io/icelake/releases/tag/v0.0.8 has been released.

File: src/query/storages/iceberg/src/table.rs [hunk 1]
```diff
@@ -68,15 +68,11 @@ impl IcebergTable {
         tbl_root: DataOperator,
     ) -> Result<IcebergTable> {
         let op = tbl_root.operator();
-        let mut table = icelake::Table::new(op.clone());
-        table
-            .load()
+        let table = icelake::Table::open_with_op(op.clone())
```
Comment (youngsofun) [inline src/query/storages/iceberg/src/table.rs:71]: @Xuanwo the icelake api changed, plz review

**Pull request merged**